### PR TITLE
Add a caterpillar plot and RanefInfo struct

### PR DESCRIPTION
**Pull request by dmbates**
- needs tests

### CODE REVIEW

File: src/caterpillar.jl [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                re.levels,
+                Matrix(adjoint(eff)),
+                Matrix(adjoint(dropdims(sqrt.(mapslices(diag, cv, dims=1:2)), dims=2))),
+            )
+        )
+    end
+    NamedTuple{fn}((val...,))
+end
+
+function caterpillar!(f::Figure, r::RanefInfo; orderby=1)
```
Comment (palday) [inline src/caterpillar.jl:37]: it took me a few minutes to understand this, but `orderby` is just telling it which of the random effects to sort by and defaults to the first (so usually the intercept), correct?

File: src/caterpillar.jl [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    y = axes(rr, 1)
+    cn = r.cnames
+    axs = [Axis(f[1, j]) for j in axes(rr, 2)]
+    for (j, ax) in enumerate(axs)
+        xvals = view(rr, ord, j)
+        scatter!(ax, xvals, y, color=(:red, 0.2))
+        errorbars!(ax, xvals, y, view(r.stddev, ord, j), direction=:x)
+        ax.xlabel = cn[j]
+        ax.yticks = y
+        j > 1 && hideydecorations!(ax, grid=false)
```
Comment (palday) [inline src/caterpillar.jl:50]: a `linkyaxes!` call somewhere in here would probably also be good

Comment (dmbates) [reply src/caterpillar.jl:59]: Thanks for the suggestion.  Now added.

File: src/caterpillar.jl (hunk 1 above)
Comment (dmbates) [reply src/caterpillar.jl:45]: I added more documentation.

**Comment by dmbates** [comment]
I also added some utility functions for creating an `xyplot`-like multi-panel data plot.

**Comment by dmbates** [comment]
I exported a couple of examples from the `xyplot.jl` file just to have them available for a posting on slack.  They probably shouldn't stay in this package but this is all WIP.

File: src/xyplot.jl [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+"""
+    simplelinreg(x, y)
+
+Return the coefficients, `[a, b]`,  from a simple linear regression, `y = a + bx + ϵ`
+"""
+function simplelinreg(x, y)
+    A = cholesky!(Symmetric([length(x) sum(x) sum(y); 0.0 sum(abs2, x) dot(x, y); 0.0 0.0 sum(abs2, y)])).factors
+    ldiv!(UpperTriangular(view(A, 1:2, 1:2)), view(A, 1:2, 3))
+end
```
Comment (palday) [inline src/xyplot.jl:46]: I think I probably would have just used the fact `A \ b` uses the QR decomposition instead of doing all this. :smile:

**Review by palday — APPROVED** [review]

File: src/xyplot.jl (hunk 3 above)
Comment (dmbates) [reply src/xyplot.jl:46]: You never know - you start saving a dozen allocations and a few microseconds here and there and pretty soon you have really fast code. :smiley:

Comment (palday) [reply src/xyplot.jl]: Between you and @araslan, I get a weird tingle whenever I realize I could save an allocation, no matter how tiny.

File: src/xyplot.jl [hunk 4]
```diff
@@ -0,0 +1,46 @@
+"""
+    Clevelandaxes!(f::Figure, labs, layout)
+
+Create a set of axes within `f` with rows, columns determined by `layout` sufficient to hold `labs`
+"""
+function Clevelandaxes!(f::Figure, labs, layout)
```
[the remark this answers is not in the document]
Comment (ararslan) [reply src/xyplot.jl]: Cleveland Axes!

**Pull request merged**